Source/WebCore/platform/graphics/filters/FilterEffect.cpp

```diff
@@ -147,7 +157,8 @@ RefPtr<FilterImage> FilterEffect::apply(const Filter& filter, FilterImage& input
 
 RefPtr<FilterImage> FilterEffect::apply(const Filter& filter, const FilterImageVector& inputs, FilterResults& results, const std::optional<FilterEffectGeometry>& geometry)
 {
-    ASSERT(inputs.size() == numberOfImageInputs());
+    if (inputs.size() != numberOfImageInputs())
```

Comment:
I wonder if it's worth logging here, instead of `takeImageInputs`. I guess there `takeImageInputs` is the interesting place to notify a developer something is wrong, so this seems fine.